src/plugins/discover/public/application/discover_router.tsx

```diff
@@ -23,12 +25,25 @@ import type { DiscoverCustomizationContext } from '../customizations';
 
 export interface DiscoverRouterProps {
   services: DiscoverServices;
-  customizationContext: DiscoverCustomizationContext;
+  customizationContext$: Observable<DiscoverCustomizationContext>;
   experimentalFeatures: ExperimentalFeatures;
   history: History;
 }
 
-export const DiscoverRouter = ({ services, history, ...routeProps }: DiscoverRouterProps) => {
+export const DiscoverRouter = ({
+  customizationContext$,
+  services,
+  history,
+  ...routeProps
+}: DiscoverRouterProps) => {
+  const customizationContext = useObservable(customizationContext$);
+
+  // The Discover state is not reactive, so we must avoid initialization before
+  // the first value is emitted
+  if (customizationContext == null) {
+    return null;
+  }
```

Comment:
That's true, and I think that's what would actually happen since presumably `useObservable` needs to call `setState` after its observable is triggered. I suppose we know `customizationContext$` will always fire anyway and we won't get stuck rendering `null`, so it's not a big concern. We can always consider alternatives as part of the One Discover architecture changes, and the only real impact is that we'll probably need to update the Jest tests to wait for the full render now.